FWCore/Framework/src/TransformerBase.cc

```diff
@@ -48,15 +61,51 @@ namespace edm {
     }
   }
 
-  void TransformerBase::transformImp(std::size_t iIndex,
-                                     ProducerBase const& iBase,
-                                     edm::EventForTransformer& iEvent) const {
-    auto handle = iEvent.get(transformInfo_.get<1>(iIndex), transformInfo_.get<0>(iIndex));
+  void TransformerBase::transformImpAsync(edm::WaitingTaskHolder iHolder,
+                                          std::size_t iIndex,
+                                          ProducerBase const& iBase,
+                                          edm::EventForTransformer& iEvent) const {
+    if (transformInfo_.get<3>(iIndex)) {
+      std::optional<decltype(iEvent.get(transformInfo_.get<1>(iIndex), transformInfo_.get<0>(iIndex)))> handle;
+      try {
+        handle = iEvent.get(transformInfo_.get<1>(iIndex), transformInfo_.get<0>(iIndex));
+      } catch (...) {
+        iHolder.doneWaiting(std::current_exception());
+        return;
+      }
+      if (handle->wrapper()) {
+        auto cache = std::make_shared<std::any>();
```

Comment:
I _believe_ in principle the `cache` could be also `std::shared_ptr<void>` (in which case the `TransformBase::PreTransformFunction` should probably return the `std::shared_ptr<void>`). I think `shared_ptr<void>` would allow supporting move-only types (assuming the copyability of `shared_ptr` itself is sufficient), and it would avoid the (possibly minuscule) run-time cost of `std::any_cast`. I _believe_ the code around is sufficient to guarantee type safety even with `void`.

Reply:
Nevermind, I figured it out why my suggestion above would not work. The space (object) shared between the "acquire" and "produce" parts really needs to be created here.